feat(new-trace): Adding transaction quota exceeded banner

Abdkhan14 opened the pull request:
Note that the billing checkout links will only work in prod:
<img width="781" alt="Screenshot 2024-08-04 at 11 10 18 PM" src="https://github.com/user-attachments/assets/2844a067-64ec-4bae-8a76-fe734d8fb377">


Logic for rendering banner:
- For an errors only trace
- If transactions were dropped around the time a trace occured and
- If the org still has transactions quota maxed out. 

Trace for testing (must remove feature flag check highlighted in the pr comments below): [link](https://kreios.dev.getsentry.net:7999/performance/trace/142e6c398dc44969ae673e9671f9f58d/?eventId=0c834757e75540b5ae09fc1c63374c6b&groupId=5625634392&pageEnd=2024-07-31T09%3A04%3A13.838&pageStart=2024-07-30T09%3A04%3A13.838&query=is%3Aunresolved+issue.priority%3A%5Bhigh%2C+medium%5D&referrer=issue-stream&source=issue_details&stream_index=2&timestamp=1722373453)

Commit:
feat(new-trace): Adding trace banner for generic trouble shooting.

Commit:
feat(new-trace): Simplifying file structure

Commit:
feat(new-trace): Minor iterations.

edwardgou-sentry commented on static/app/views/performance/newTraceDetails/traceTypeWarnings/styles.tsx:
```diff
@@ -0,0 +1,169 @@
+          </ActionButton>
+          <ActionButton>
+            <Button
+              onClick={() => props.onSecondaryButtonClick()}
```
```suggestion
              onClick={props.onSecondaryButtonClick}
```

Commit:
feat(new-trace): Minor iteration.

edwardgou-sentry commented on static/app/views/performance/newTraceDetails/traceTypeWarnings/styles.tsx:
```diff
@@ -0,0 +1,169 @@
+          </ActionButton>
+        </ButtonsWrapper>
+      </ActionsWrapper>
+      {<BannerBackground image={props.image} />}
```
```suggestion
      <BannerBackground image={props.image} />
```

edwardgou-sentry commented on static/app/views/performance/newTraceDetails/traceTypeWarnings/styles.tsx:
```diff
@@ -0,0 +1,169 @@
+          {
+            key: 'dismiss',
+            label: t('Dismiss'),
+            onAction: () => {
+              dismiss();
+            },
```
```suggestion
            onAction: dismiss,
```

edwardgou-sentry commented on static/app/views/performance/newTraceDetails/traceTypeWarnings/styles.tsx:
```diff
@@ -0,0 +1,169 @@
+          {
+            key: 'snooze',
+            label: t('Snooze'),
+            onAction: () => {
+              snooze();
+            },
```
```suggestion
            onAction: snooze,
```

edwardgou-sentry commented on static/app/views/performance/newTraceDetails/traceTypeWarnings/useTransactionUsageStats.tsx:
```diff
@@ -0,0 +1,67 @@
+import type {TraceTree} from '../traceModels/traceTree';
+
+// 1 hour in milliseconds
+const ONE_HOUR = 3600 * 1000;
```
```suggestion
const ONE_HOUR = 60 * 60 * 1000;
```
nit: this feels slightly more readable to me 😄

edwardgou-sentry reviewed (approved):
left a few nits!

Abdkhan14 committed:
fix(new-trace): Ensuring that projects from trace are prioritized in the checklist dropdown. (#75349)

Trace for testing:
[link](https://perfquotaorg.dev.getsentry.net:7999/performance/trace/dcbf0cf7bbd5425bbea1116b8f7d8aa5/?project=4506836555595776#)

The checklist wasn't pushing projects from the trace to the top of the
dropdown:
<img width="1504" alt="Screenshot 2024-07-31 at 12 53 30 PM"
src="https://github.com/user-attachments/assets/b7df0357-728c-4275-9932-de01866d2256">

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

Abdkhan14 committed:
Update static/app/views/performance/newTraceDetails/traceTypeWarnings/styles.tsx

Co-authored-by: edwardgou-sentry <83961295+edwardgou-sentry@users.noreply.github.com>

Abdkhan14 committed:
Update static/app/views/performance/newTraceDetails/traceTypeWarnings/styles.tsx

Co-authored-by: edwardgou-sentry <83961295+edwardgou-sentry@users.noreply.github.com>

Abdkhan14 committed:
Update static/app/views/performance/newTraceDetails/traceTypeWarnings/styles.tsx

Co-authored-by: edwardgou-sentry <83961295+edwardgou-sentry@users.noreply.github.com>

Abdkhan14 committed:
Update static/app/views/performance/newTraceDetails/traceTypeWarnings/styles.tsx

Co-authored-by: edwardgou-sentry <83961295+edwardgou-sentry@users.noreply.github.com>

Abdkhan14 committed:
Update static/app/views/performance/newTraceDetails/traceTypeWarnings/useTransactionUsageStats.tsx

Co-authored-by: edwardgou-sentry <83961295+edwardgou-sentry@users.noreply.github.com>

Commit:
feat(new-trace): Updating text.

Commit:
feat(new-trace): Updated text
